src/connector/src/sink/postgres.rs

```diff
@@ -338,201 +355,219 @@ impl PostgresSinkWriter {
+            match item {
+                LogStoreReadItem::StreamChunk { chunk, chunk_id } => {
+                    self.write_batch(chunk).await?;
+                    log_reader.truncate(TruncateOffset::Chunk { epoch, chunk_id })?;
+                }
+                LogStoreReadItem::Barrier { .. } => {
+                    log_reader.truncate(TruncateOffset::Barrier { epoch })?;
+                }
+                LogStoreReadItem::UpdateVnodeBitmap(_) => {}
+            }
+        }
+    }
 }
 
-fn create_upsert_sql(schema: &Schema, table_name: &str, pk_indices: &[usize]) -> String {
+fn create_insert_sql(schema: &Schema, table_name: &str, number_of_rows: usize) -> String {
+    let number_of_columns = schema.len();
     let columns: String = schema
         .fields()
         .iter()
         .map(|field| field.name.clone())
         .collect_vec()
         .join(", ");
-    let parameters: String = (0..schema.fields().len())
-        .map(|i| format!("${}", i + 1))
-        .collect_vec()
-        .join(", ");
-    let pk_columns = pk_indices
-        .iter()
-        .map(|i| schema.fields()[*i].name.clone())
-        .collect_vec()
-        .join(", ");
-    let update_parameters: String = (0..schema.fields().len())
-        .filter(|i| !pk_indices.contains(i))
+    let parameters: String = (0..number_of_rows)
         .map(|i| {
-            let column = schema.fields()[i].name.clone();
-            let param = format!("${}", i + 1);
-            format!("{column} = {param}")
+            let row_parameters = (0..number_of_columns)
+                .map(|j| format!("${}", i * number_of_columns + j + 1))
+                .collect_vec()
+                .join(", ");
+            format!("({row_parameters})")
         })
         .collect_vec()
         .join(", ");
-    format!("INSERT INTO {table_name} ({columns}) VALUES ({parameters}) on conflict ({pk_columns}) do update set {update_parameters}")
+    format!("INSERT INTO {table_name} ({columns}) VALUES {parameters}")
 }
 
-fn create_delete_sql(schema: &Schema, table_name: &str, pk_indices: &[usize]) -> String {
-    let parameters: String = pk_indices
-        .iter()
-        .map(|i| format!("{} = ${}", schema.fields()[*i].name, i + 1))
+fn create_delete_sql(
+    schema: &Schema,
+    table_name: &str,
+    pk_indices: &[usize],
+    number_of_rows: usize,
+) -> String {
+    let number_of_pk = pk_indices.len();
+    let parameters: String = (0..number_of_rows)
+        .map(|i| {
+            let row_parameters: String = pk_indices
+                .iter()
+                .map(|j| {
+                    format!(
+                        "{} = ${}",
+                        schema.fields()[*j].name,
+                        i * number_of_pk + j + 1
+                    )
+                })
+                .collect_vec()
+                .join(" AND ");
+            format!("({row_parameters})")
+        })
         .collect_vec()
-        .join(" AND ");
+        .join("OR");
```

Comment:
Maybe we can use `PK In (...)` instead of `PK = xxx or PK = yyy` which is more compacted and easy to be recognized by the external system.

Reply:
Improve later. Merge to let user try it. https://github.com/risingwavelabs/risingwave/issues/19913